src/linear_solver_routines/arnoldi_solver.jl

```diff
@@ -0,0 +1,71 @@
+# Date 09/10/2024
+# Author: Christian Varner
+# Purpose: Implement deterministic solver.
+
+"""
+    arnoldi!(x, A, b, k)
+
+Finds an approximate solution to the linear system `Ax = b` by doing an Arnoldi Iteration
+to build an orthonormal basis for the Krylov Subspace of size `k`. 
+    
+The method was detailed in
+
+Timsit, Grigori, and Balabanov. "Randomized Orthogonal Projection Methods for Krylov Subspace Solvers."
+https://arxiv.org/abs/2302.07466.
+
+# Parameters
+- `x::AbstractVector`, initial solution to the linear system `Ax = b`. Will be overwritten.
+- `A::AbstractMatrix`, coefficient matrix for linear system.
+- `b::AbstractVector`, constant vector for linear system.
+- `k::Int`, number of iterations to do (size of Krylov Subspace).
+"""
+function arnoldi!(
+    x::AbstractVector,
+    A::AbstractMatrix,
+    b::AbstractVector,
+    k::Int
+)
+
+    # helper function to form and solve the linear system after main loop
+    function solve_linear_system!(
+        x::AbstractVector, 
+        beta::Float64, 
+        H::AbstractMatrix, 
+        V::AbstractMatrix, 
+        k::Int64)
+
+        y = zeros(k)
+        y[1] = beta
+        z = H[1:k, 1:k] \ y
+        x .+= V[:, 1:k] * z
+    end
```

Comment:
This can be a utility function defined outside of `arnoldi!`. It should come with its own docstring and maybe a more specific name.